### PR TITLE
ci: add coverity scans

### PR DESCRIPTION
**Pull request by aneshlya**
Our Coverity open source project is set up here: https://scan.coverity.com/projects/intel-ispc?tab=overview.
Open source Coverity has limitations on number of runs: https://scan.coverity.com/faq#frequency so I'm setting it to run by schedule or on workflow_dispatch.
Here is example run: https://github.com/aneshlya/ispc/actions/runs/9998651864

Feel free to contribute to this PR if you think something is missing.

### CODE REVIEW

File: .github/workflows/scripts/run-coverity.sh [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+echo "Adding Coverity to PATH..."
+TOOL_DIR=$(find "$TOOL_BASE" -type d -name 'cov-analysis*')
+export PATH="$TOOL_DIR/bin:$PATH"
+
+# Clean up the build directory
+cd "$GITHUB_WORKSPACE"
+cmake --build build --target clean
+
+echo "Running Coverity analysis..."
```
Comment (nurmukhametov) [inline .github/workflows/scripts/run-coverity.sh:70]: In my opinion, this comment is slightly misleading. The analysis occurs on the remote Coverity server, which is why there are strict limits on the number of runs. Here, the ISPC build is captured along with the source code information.

**Review by nurmukhametov — APPROVED** [review]
LGTM!

**Comment by dbabokin** [comment]
Should we add the condition to the workflow that it's the main repo, but not the fork?

File: .github/workflows/scripts/run-coverity.sh (hunk 1 above)
Comment (aneshlya) [reply .github/workflows/scripts/run-coverity.sh]: Fixed

**Comment by aneshlya** [comment]
> Should we add the condition to the workflow that it's the main repo, but not the fork?

Added check

**Pull request merged**